src/Web/HTML/Common.purs

```diff
@@ -0,0 +1,30 @@
+module Web.HTML.Common where
+
+import Prelude
+
+import Data.Newtype (class Newtype)
+
+-- | A type-safe wrapper for property names.
+-- |
+-- | The phantom type `value` describes the type of value which this property
+-- | requires.
+newtype PropName value = PropName String
+
+derive instance newtypePropName :: Newtype (PropName value) _
+derive newtype instance eqPropName :: Eq (PropName value)
+derive newtype instance ordPropName :: Ord (PropName value)
+
+-- | A type-safe wrapper for attribute names.
+newtype AttrName = AttrName String
+
+derive instance newtypeAttrName :: Newtype AttrName _
+derive newtype instance eqAttrName :: Eq AttrName
+derive newtype instance ordAttrName :: Ord AttrName
+
+-- | A wrapper for strings which are used as CSS classes.
+newtype ClassName = ClassName String
+
+derive instance newtypeClassName :: Newtype ClassName _
+derive newtype instance eqClassName :: Eq ClassName
+derive newtype instance ordClassName :: Ord ClassName
+derive newtype instance semigroupClassName :: Semigroup ClassName
```

Comment:
```suggestion
derive newtype instance semigroupClassName :: Semigroup ClassName

```

Reply:
I still have a problem with this instance - I can see that argument for wanting to support `"class1 class2"` as considered to be a class name, as unless we actually make a smart constructor that rejects names with spaces there's no guarantee it's happening, but now there's a problem with this meaning of this instance too IMO:

If I have `x :: ClassName` and `y :: ClassName` it seems like the result of that should be `ClassName (unwrap x <> " " <> unwrap y)`. The name `ClassName` suggests to me that the value is not part of a class name, but it's a whole class name, so by combining them without a space it's constructing a _new_ class name, not combining two class names.

But I can also see why that might be unexpected for some people, which is kinda why I don't like the instance.

Reply:
Yes, that's a good point, and it's the reason why in my own applications I have a separate `classes` function which concatenates with spaces (unlike a standard semigroup instance, which concatenates without spaces (rightly so)). I agree that we shouldn't have a semigroup instance as the law-abiding one is not what you probably want.

Reply:
It would still be law abiding if it included the space, it's just that I think either instance we choose has the potential to trip people up, depending on how they see the type ("it's just a newtype string wrapper" vs a thing with its own semantics).

Reply:
Well, now I can't remember why I thought that was a semigroup law; I must be thinking of the left/right identity laws for monoid.

For what it's worth, we _could_ assert that this is a thing with its own semantics, where `ClassName` represents a distinct CSS class, and where append includes a separating space. I don't think I've had a conversation with someone using this type in the Halogen world where they saw it as anything else, though that's a small anecdote.

Regardless, if I'm overconfident and the camps are split then it's much better not to have an instance.

Reply:
It arose here a while back: https://github.com/purescript-halogen/purescript-halogen/pull/512 🙁

Reply:
Yea, I say we take it out to avoid the confusion.

Reply:
Either instance is a perfectly fine semigroup; semigroups only need associativity, and the space-adding version’s associativity follows from normal string concatenation being associative. The difference is that if you use the space-adding Semigroup, you can’t have a Monoid instance because there’s no identity element.

Reply:
> I say we take it out to avoid the confusion.

Will do. For reference here is where the instance was added for Halogen https://github.com/purescript-halogen/purescript-halogen/issues/451